tabgen/src/gen.rs

```diff
@@ -0,0 +1,495 @@
+//! The basic idea is that we segment codepoints into one of a,
```

Comment:
yep this one is wrong too. also, i'm aware theres too much redundant stuff in this file, it can be simplified.